x-pack/plugins/fleet/server/services/setup.ts

```diff
@@ -230,3 +232,58 @@ export function formatNonFatalErrors(
     }
   });
 }
+
+/**
+ * Ensures that "managed" packages are using the latest comptabile version. In cases where
+ * Kibana has been upgraded or rolled back, we need to ensure the versions of packages that are
+ * installed by default or designated as "auto update" are aligned with the stack version.
+ */
+export async function ensureManagedPackageVersions(
```

Comment:
@kpollich do we still need to split this out into a PR separate from this testing PR?

Reply:
I don't think so, actually. We verified that everything works as expected with package versions across upgrades/rollbacks in our manual process here: https://github.com/elastic/kibana/issues/111858#issuecomment-986970512

The QAS folks also verified on their end that package versions work as expected on upgrade/rollback in their own test ticket here: https://github.com/elastic/kibana/issues/120726

So, I think this `ensureManagedPackageVersions` call where we explicitly iterate through packages and do version checks is redundant. We can close this WIP PR out.